src/main/java/org/example/services/AuthService.java

```diff
@@ -0,0 +1,41 @@
+package org.example.services;
+
+import io.jsonwebtoken.Jwts;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import org.example.daos.interfaces.IAuthDao;
+import org.example.exception.AuthDaoException;
+import org.example.exception.InvalidCredentialsException;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+
+import java.security.Key;
+import java.util.Date;
+
+@Getter @Setter @AllArgsConstructor
+public class AuthService {
+    private final IAuthDao authDao;
+    private final Key key;
+
+    public String login(final LoginRequest loginRequest) throws
+            InvalidCredentialsException, AuthDaoException {
+        User user = authDao.getUser(loginRequest);
+        if (user.getUsername() == null || user.getPassword() == null) {
```

Comment:
Should this be 2 separate if statements, one for when the username does exist in the db but the user gave the wrong password, and one for when the username and password don't exist? Or is that front end logic I'm not really sure haha

Reply:
That's a valid point. Although since it only matters if the service will be either returning a jwt token or not, in either case (if it was a valid username or password) the controller consumes that information in the same way (by returning status code of 400). We could somehow distinguish if it was a wrong username or password, but I'm thinking it may be more secure if we don't let the client know either (in the case that it is a hacker trying to verifying if they found a valid username in the system).

Bit of a ramble, but let me know if my logic makes sense or not.

Reply:
So if we use optionals this will need to be updated. You can do user.isPresent()

Reply:
Updated!